crates/bevy_tasks/src/task_pool.rs

```diff
@@ -557,6 +583,71 @@ impl TaskPool {
+    /// future without yielding is not okay, as it may prevent the task pool
+    /// from driving other futures forward. This function runs the provided
+    /// closure on a thread dedicated to blocking operations.
+    ///
+    /// This call will spawn more blocking threads when they are requested
+    /// through this function until the upper limit configured.
+    /// This limit is very large by default (500), because `spawn_blocking` is often
+    /// used for various kinds of IO operations that cannot be performed
+    /// asynchronously. When you run CPU-bound code using `spawn_blocking`,
+    /// you should keep this large upper limit in mind; to run your
+    /// CPU-bound computations on only a few threads. Spawning too many threads
+    /// will cause the OS to [thrash], which may impact the performance
+    /// of the non-blocking tasks scheduled onto the `TaskPool`.  
+    ///
+    /// Closures spawned using `spawn_blocking` cannot be cancelled. When you shut down the executor, it will wait
+    /// indefinitely for all blocking operations to finish.
+    ///
+    /// [thrash]: https://en.wikipedia.org/wiki/Thrashing_(computer_science)
+    pub fn spawn_blocking<T>(&self, f: impl FnOnce() -> T + Send + 'static) -> Task<T>
+    where
+        T: Send + 'static,
+    {
+        Task::new(unblock(f))
+    }
+
+    /// Spawns a static future onto on a thread where blocking is acceptabl.e
+    /// The returned [`Task`] is a future that can be polled for the result.
+    /// It can also be canceled and "detached", allowing the task to continue
+    /// running even if dropped. In any case, the pool will execute the task
+    /// even without polling by the end-user.
+    ///
+    /// This function is equivalent to calling `task_pool.spawn_blocking(|| block_on(f))`.
+    ///
+    /// If the future is expected to terminate quickly, or will not spend a
+    /// signficant amount of time performing blocking CPU-bound or IO-bound
+    /// operations, [`spawn`] should be used instead. The ideal use case for
+    /// this function is for launching a future that may involve a combination
+    /// of async IO and blocking operations (i.e. loading large scenes).
+    ///
+    /// This call will spawn more blocking threads when they are requested
+    /// through this function until the upper limit configured.
+    /// This limit is very large by default (500), because `spawn_blocking` is often
+    /// used for various kinds of IO operations that cannot be performed
+    /// asynchronously. When you run CPU-bound code using `spawn_blocking`,
+    /// you should keep this large upper limit in mind; to run your
+    /// CPU-bound computations on only a few threads. Spawning too many threads
+    /// will cause the OS to [thrash], which may impact the performance
+    /// of the non-blocking tasks scheduled onto the `TaskPool`.  
+    ///
+    /// Closures spawned using `spawn_blocking` cannot be cancelled. When you
```

Comment:
I think it'll work to cancel tasks at await points with the `Task`

Reply:
Fixed.